### DISCUSSION TITLE
Unable to figure out how to use aggregate queries in firestore

### DISCUSSION THREAD

**Discussion by ayushbgl** [invertase/react-native-firebase#7634] [General]
I want to use `sum` function as described in [this link](https://firebase.google.com/docs/firestore/query-data/aggregation-queries). I have a query similar to this.

```
const coll = firestore.collection('cities');
const sumAggregateQuery = coll.aggregate({
         totalPopulation: AggregateField.sum('population'),
       });
  
const snapshot = await sumAggregateQuery.get();
console.log('totalPopulation: ', snapshot.data().totalPopulation);
```

I was able to make count query work but not sum query. I am unable to import AggregateField from '@react-native-firebase/firestore'.  Please let me know if there's any documentation or working example to figure out how to use aggregations (sum and average) in my app. Thanks

**Comment by mikehardy** [comment]
Hi there! With apologies, we have not implemented the sum or average aggregate queries yet

I would personally love to do so and they are on my list but my list is quite long and I'm not sure when I'll have time - it could be more on the "months from now" time scale vs "soon".

If this is an important use case for you I'd be more than happy to collaborate on a PR that got them in though, if you wanted to try to create a PR.

Now that we have the first aggregate query in, adding more shouldn't be nearly as involved - you can see the original PR that added 

https://github.com/invertase/react-native-firebase/pull/6616/commits/e9f8968981edf8ce2b92900913640d9fba6b238c

- the aggregate query infrastructure (only has to be done once! already done)
- the count feature

You might be able to look at that and pattern an addition of sum off it for a relatively quick PR ?

**Comment by nicolaosm** [comment]
We're also looking to use .avg() and addAggregation as well but these awesome features are missing?

Would really appreciate if those are added in upcoming updates! Thanks

**Comment by RuarddeBruyn** [comment]
Any progress with the above? Seems to be "years from now" vs soon.
